Makefile

```diff
@@ -0,0 +1,36 @@
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	cargo clippy -- -D warnings
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
+
+# --- testing -------------------------------------------------------------------------------------
+
+.PHONY: test-build
+test-build: ## Build the test binary
```

Comment:
IMO we don't need this task at all, but I'm fine with keeping it for consistency with other projects for those that use it in their workflow. In any case, we can remove the explicit `--cargo-profile` flag here (since the `test-dev` profile should be removed)